### PR TITLE
Add Config API reference for kubeconfig

### PR DESCRIPTION
**Pull request by tengqm**
The file added is generated using kubernetes-sigs/reference-docs/genref tool, with logics and configs improved by
https://github.com/kubernetes-sigs/reference-docs/pull/309 and https://github.com/kubernetes-sigs/reference-docs/pull/310.

related: #38478

### CODE REVIEW

**Comment by tengqm** [comment]
Problem resolved.
/hold cancel

**Comment by sftim** [comment]
A very welcome change!

/approve

File: content/en/docs/reference/config-api/kubeconfig.v1.md [hunk 1]
```diff
@@ -0,0 +1,603 @@
+
```
[the remark this answers is not in the document]
Comment (tengqm) [reply content/en/docs/reference/config-api/kubeconfig.v1.md:1]: Tried hard to kill it ... you know I'm a perfectionist ... sigh

[the remark this answers is not in the document]
Comment (tengqm) [reply content/en/docs/reference/config-api/kubeconfig.v1.md]: done

**Pull request merged**